.github/workflows/mobile_job.yml

```diff
@@ -0,0 +1,140 @@
+name: Run mobile tests on devices
+
+on:
+  workflow_call:
+    inputs:
+      job-name:
+        description: Name for the job, which is displayed in the GitHub UI
+        default: mobile-job
+        type: string
+      device-type:
+        description: The type of device (iOS or Android) to test against
+        type: string
+      runner:
+        description: The runner to run the test on
+        type: string
+      timeout:
+        description: Timeout for the job (in minutes)
+        required: false
+        default: 60
+        type: number
+      python-version:
+        description: Set the python version used in the job
+        required: false
+        type: string
+        default: '3.11'
+
+      # AWS Device Farm, this can be copied from AWS console and it's default to
+      # PyTorch project
+      project-arn:
+        description: The AWS Device Farm project where the test runs
+        default: 'arn:aws:devicefarm:us-west-2:308535385114:project:b531574a-fb82-40ae-b687-8f0b81341ae0'
+        type: string
+
+      # Pulling test-infra itself for device farm runner script
+      test-infra-repository:
+        description: Test infra repository to use
+        default: 'pytorch/test-infra'
+        type: string
+      test-infra-ref:
+        description: Test infra reference to use
+        default: ''
+        type: string
+
+      # iOS-specific inputs
+      ios-ipa-archive:
+        description: The link to the iOS app IPA archive to run the tests
+        required: false
+        type: string
+        default: ''
+      ios-xctest-zip:
+        description: The link to the iOS test suite itself
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  job:
+    name: ${{ inputs.job-name }} (${{ inputs.device-type }})
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: ${{ inputs.timeout }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Clean workspace
+        run: |
+          echo "::group::Cleanup debug output"
+          rm -rfv "${GITHUB_WORKSPACE}"
+          mkdir -p "${GITHUB_WORKSPACE}"
+          echo "::endgroup::"
+
+      - name: Authenticate with AWS
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          # TODO: Replace with an actual role from pytorch-gha-infra
+          role-to-assume: arn:aws:iam::308535385114:role/DEBUG-DEVICE-FARM-ROLE
```

Comment:
This is to be replaced by TODO

Reply:
We will also probably want to make a follow up issue to ensure this doesn't break if / when we migrate to ARC

Reply:
The credentials are acquired at user/workflow level. Given that ARC provides the same version of AWS cli/toolset I believe that there are no changes required. Unless there are permissions granted to the node role that won't be available in ARC. But I've been disencouraging it for a while, so I believe we don't have such cases ATM